rust/crd/src/lib.rs

```diff
@@ -179,6 +179,36 @@ pub struct HdfsClusterConfig {
     /// Deprecated, please use `.spec.nameNodes.config.listenerClass` and `.spec.dataNodes.config.listenerClass` instead.
     #[serde(default)]
     pub listener_class: DeprecatedClusterListenerClass,
+
+    /// Configuration to control HDFS topology (rack) awareness feature
+    pub rack_awareness: Option<Vec<TopologyLabel>>,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Eq, Hash, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TopologyLabel {
+    /// Name of the label type that will be typically either `node` or `pod`, used to create a
+    /// topology out of datanodes.
+    label_type: TopologyLabelType,
+
+    /// Name of the label that will be used to resolve a datanode to a topology zone.
+    label_name: String,
+}
```

Comment:
Pending https://github.com/stackabletech/decisions/issues/2

Reply:
I think this should be independent of stackabletech/decisions#2 as that is a much smaller piece of work and we want to ship this feature (rack awareness) with the next release.

Reply:
Ideally of course we don't want a breaking change, but we had already agreed on the original structure at a recent arch meeting.

Reply:
Ah ok, no problem. Will mark as resolved.